FirebaseStorageSwift/Sources/AsyncAwait.swift

```diff
@@ -66,12 +70,33 @@ import FirebaseStorage
     func putFileAsync(from url: URL,
                       metadata: StorageMetadata? = nil) async throws -> StorageMetadata {
       typealias MetadataContinuation = CheckedContinuation<StorageMetadata, Error>
-      return try await withCheckedThrowingContinuation { (continuation: MetadataContinuation) in
-        // TODO: Use task to handle progress and cancellation.
-        _ = self.putFile(from: url, metadata: metadata) { result in
-          continuation.resume(with: result)
+
+      let metadata =
+        try await withCheckedThrowingContinuation { (continuation: MetadataContinuation) in
```

Comment:
There is `withTaskCancellationHandler` but I'm not sure how to combine that with `withCheckedThrowingContinuation` and not sure it would work with cancelling the Objective C task anyway.

https://forums.swift.org/t/how-to-cancel-a-publisher-when-using-withtaskcancellationhandler/49688/5